packages/components/src/Typeahead/Typeahead.component.renderers.js

```diff
@@ -134,6 +134,9 @@ function emphasise(text, value) {
 	if (!value) {
 		return [text];
 	}
+	if (!text) {
```

Comment:
Can you put that before the test on `value` ?
Or else if there is no text and no value, it will return `[undefined]`. This is well supported in our use but much cleaner if it returns empty string.